### PR TITLE
Feature/bael 6101 singleton

### PR DESCRIPTION
**Pull request by dstr89**

### CODE REVIEW

File: spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    private static volatile SingletonPattern instance = null;
+
+    private SingletonPattern() {}
+
+    public static SingletonPattern getInstance() {
+        if (instance == null) {
+            synchronized(SingletonPattern.class) {
+                if (instance == null) {
+                    instance = new SingletonPattern();
```
Comment (ashleyfrieze) [inline spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java:13]: I'd be interested in comparing this to 

```java
public final class SingletonPattern {
   private static final SingletonPattern instance = new SingletonPattern();

   public static SingletonPattern getInstance() { return instance; }
}
```

I believe that these are, functionally, the same!!!

File: spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.baeldung.sample.singleton;
+
+public final class SingletonPattern {
```
Comment (ashleyfrieze) [inline spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java:3]: The class name - `SingletonPattern` - might be worth improving:

`ThreadSafeSingleInstance`?

**Review by ashleyfrieze — CHANGES_REQUESTED** [review]

File: spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java (hunk 1 above)
Comment (dstr89) [reply spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java:13]: Hi @ashleyfrieze, functionally I would say yes. But your example is not using lazy loading, which is described in the article. I would leave it like this.

File: spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java (hunk 2 above)
Comment (dstr89) [reply spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java:3]: Makes sense, fixed.

**Review by andrzejsydor — APPROVED** [review]
What about enum as a Singleton?

**Pull request merged**